src/south/MQTT/MQTT.class.js

```diff
@@ -41,6 +43,8 @@ class MQTT extends ProtocolHandler {
     this.username = username
     this.password = Buffer.from(this.encryptionService.decryptText(password))
     this.qos = qos
+    this.persistent = persistent
+    this.clientId = clientId || `OIBus-${Math.random().toString(16).substr(2, 8)}`
```

Comment:
our nodejs supports the ?? notation..

Reply:
**??** is the nullish coalescing operator which will only use the value from the right side when the value on left side is null or undefined. It means that if `clientId` is an empty string, it will use the empty string which is not valid for MQTT.
Using **||** will ensure that `this.clientId` is not an empty string, because it will use the value from the right side if the value on the left side is falsy.

Reply:
ok, my mistake.